### PR TITLE
Removed extension cli project

### PR DESCRIPTION
**Pull request by SpencerWBarnes**

### CODE REVIEW

**Review by calebmshafer — APPROVED** [review]

**Comment by calebmshafer** [comment]
Please add something to `docs/changehistory/NextVersion.md` that this particular client is going to be deprecated.

**Review by pmconne — APPROVED** [review]

**Comment by SpencerWBarnes** [comment]
> Please add something to `docs/changehistory/NextVersion.md` that this particular client is going to be deprecated.

Should the removal of the package config-loader also be added to the documentation?

File: docs/changehistory/NextVersion.md [hunk 1]
```diff
@@ -120,6 +120,10 @@ Several changes were made in the @bentley/ui-* packages.
 Some components in @bentley/ui-core were deprecated in favor of components in @itwinui-react.
 A few constructs were deprecated in @bentley/ui-core package with alternatives elsewhere.
 
+### @bentley/extension-cli
+
+Deprecated. It was unused and has been fully removed.
```
Comment (calebmshafer) [inline docs/changehistory/NextVersion.md:125]: Let's say something like, `The cli tool has been deprecated due to an impending change of Extensions and the Extension Service. Please continue to use the 2.x version if you still require publishing Extensions.`

**Review by calebmshafer — CHANGES_REQUESTED** [review]

**Comment by calebmshafer** [comment]
> > Please add something to `docs/changehistory/NextVersion.md` that this particular client is going to be deprecated.
> 
> Should the removal of the package config-loader also be added to the documentation?

@SpencerWBarnes yes, we probably should have put something in there for it.

File: docs/changehistory/NextVersion.md (hunk 1 above)
Comment (SpencerWBarnes) [reply docs/changehistory/NextVersion.md]: Updated doc comment

File: docs/changehistory/NextVersion.md [hunk 2]
```diff
@@ -120,6 +120,14 @@ Several changes were made in the @bentley/ui-* packages.
 Some components in @bentley/ui-core were deprecated in favor of components in @itwinui-react.
 A few constructs were deprecated in @bentley/ui-core package with alternatives elsewhere.
 
+### @bentley/extension-cli
+
+The cli tool has been deprecates due to an impending change of Extensions and the Extension Service. Please continue to use the 2.x version if you still require publishing Extensions.
```
Comment (kabentley) [inline docs/changehistory/NextVersion.md:125]: deprecates => deprecated

Comment (SpencerWBarnes) [reply docs/changehistory/NextVersion.md]: Thank you. I guess spell checkers can't spot grammar mistakes

**Review by calebmshafer — APPROVED** [review]

**Review by paulius-valiunas — APPROVED** [review]

**Comment by SpencerWBarnes** [comment]
@mergifyio update

**Pull request merged**